src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3435,6 +3421,16 @@ private static TypeWithState ApplyUnconditionalAnnotations(TypeWithState typeWit
             return typeWithState;
         }
 
+        private static TypeWithAnnotations ApplyUnconditionalAnnotations(TypeWithAnnotations declaredType, FlowAnalysisAnnotations annotations)
+        {
+            if ((annotations & FlowAnalysisAnnotations.MaybeNull) == FlowAnalysisAnnotations.MaybeNull)
+            {
+                return declaredType.AsAnnotated();
```

Comment:
>AsAnnotated [](start = 36, length = 11)

Not sure why this isn't using `MaybeDefault`.

Reply:
This particular method is generating a `TypeWithAnnotations` rather than a `TypeWithState`. The intent is to treat a `[MaybeNull]T` declaration as `T?` during analysis, even in cases where `T` cannot be annotated.

---
In reply to: [347686078](https://github.com/dotnet/roslyn/pull/39778#discussion_r347686078) [](ancestors = 347686078)